include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -656,34 +654,39 @@ struct __result_and_scratch_storage_impl : __result_and_scratch_storage_base
 #endif
     }
 
-    _T
+    auto
     __wait_and_get_value(sycl::event __event) const
     {
+        static_assert(_NResults == 1);
+
         if (is_USM())
             __event.wait_and_throw();
 
-        return __get_value();
+        if constexpr (_NResults == 1)
```

Comment:
I believe we fix here some important error that has place for every instance of `__result_and_scratch_storage` without results: we tried to load some value from memory (and so on) ever if it's absent.

@danhoeflinger, @MikeDvorskiy what do you thinks?

Reply:
Are there any places where we attempt to `__wait_and_get_value` when there is no result part of `__result_and_scratch_storage`?  It seems that in and of itself is flawed.

I agree that this is safer, but are there actually spots in the code where we are attempting to get a value that doesn't exist?

Reply:
Looking to `static_assert(_NResults == 1);` at line 660 I think it's extra condition.
Fixed (removed).